org.eclipse.winery.lsp/src/main/java/org/eclipse/winery/lsp/Server/ServerCore/DataModels/TOSCAFile.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+/**
+ * TOSCA File Record
+ *
+ * For more details on the TOSCA specification, visit:
+ * https://docs.oasis-open.org/tosca/TOSCA/v2.0/csd06/TOSCA-v2.0-csd06.html#61-keynames
+ */
+package org.eclipse.winery.lsp.Server.ServerCore.DataModels;
+
+import io.soabase.recordbuilder.core.RecordBuilder;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@RecordBuilder
+public record TOSCAFile(
+    String tosca_definitions_version,
+    Optional<String> description,
+    Optional<Map<String, Object>> metadata,
+    Optional<Object> dsl_definitions,
+    Optional<Map<String, ArtifactType>> artifact_types,
```

Comment:
Please use Java variable style: `artifactTypes`. After parsing the YAML, we are in the domain of Java, which has other variable style naming.